healthcheck.go

```diff
@@ -23,15 +25,28 @@ func (e *ErrBucketDoesNotExist) Error() string {
 	return fmt.Sprintf("Bucket %s does not exist", e.BucketName)
 }
 
-// Checker performs a check health of S3 and updates the provided CheckState accordingly
-func (s3 *S3) Checker(ctx context.Context, state *health.CheckState) error {
-	reader, err := s3.Get("s3://" + s3.BucketName)
+// Checker validates that the S3 bucket exists, and updates the provided CheckState accordingly
+func (cli *S3) Checker(ctx context.Context, state *health.CheckState) error {
+	err := cli.ValidateBucket()
 	if err != nil {
-		// errBucket := ErrBucketDoesNotExist{BucketName: s3.BucketName}
+		if awsErr, ok := err.(awserr.Error); ok {
```

Comment:
(Personal preference) I'm not a fan of having multiple indentation levels in a single method/function - I think it makes it harder to read as the scope of what the code is doing becomes more specific. 

Maybe consider extracting this error  handling logic into a separate function? Something like 
```
handleAWSErr(err awserr.Error)
```